tests/handlers/test_agent.py

```diff
@@ -30,3 +34,42 @@ def test_post_agent_unauthenticated(test_client, db_session):
 
     assert response.status_code == 403
     assert response.json()['detail'] == "Permission denied."
+
+
+def test_list_agent_empty_list(db_session, test_client, session_cookie):
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.json() == []
+
+
+def test_list_agent_success(db_session, test_client, session_cookie):
+    agent = Agent(name='test', oauth_client=OAuthClient(type='confidential'))
+    db_session.add(agent)
+    db_session.commit()
+
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.status_code == 200
+    response_body = response.json()
+
+    assert len(response_body) == 1
+
+    assert response_body[0]['name'] == 'test'
+    # assert response_body[0]['connection_type'] == 'd'
+    oauth_client = response_body[0]['oauth_client']
+    assert oauth_client['agent_id'] is not None
+    assert oauth_client['secret'] is not None
```

Comment:
Oh this is definitely not supposed to dump the secret here. You'll have to introduce more abstraction into the schemas. The `OAuthClient` should not be merged in the list call. Ideally assert this too because we really don't want it to be dumped.